sklearn/utils/tests/test_class_weight.py

```diff
@@ -26,6 +29,27 @@ def test_compute_class_weight_not_present():
     assert_raises(ValueError, compute_class_weight, "auto", classes, y)
 
 
+def test_compute_class_weight_invariance():
```

Comment:
Just double checking: make_blobs(...) yields a data set in which there are N=100 data points of which N0=50 are of class 0 and N1=50 of class 1. You're then creating 3 scenarios: one where you replicate all instances of class 0 to give a total of 150 instances of class 0 (and still 50 of class 1), one where you replicate all instances of class 1 to give a total of 150 instances of class 1 (and still 50 of class 0), and one where you replicate all instances of class 0 to give a total of 100 intsances of class 0 and similarly for class 1. 

Then you're saying: For scenario 1, according to my weighting scheme, w0 = (150 + 50) / (2 * 150) = 2/3 and w1 = (150 + 50) / (2 * 50) = 2. This gives 2/3 * 150 = 100 effective data points in class 0 and 2 * 50 = 100 effective data points in class 1. As a result, for any fixed inverse regularization parameter (C), you should get the same optimized model parameters as you'd get for scenario 3 provided you used either my weighting scheme or w0 = w1 = 1 and you used the same inverse regularization parameter C.

Similarly for scenario 2.

Yes? If so, then yes, I agree. 

And I also agree that the test will fail if you use the "auto" weighting scheme currently implemented.